KMeans.java

```diff
@@ -0,0 +1,126 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class KMeans {
```

Comment:
Every class and function (at minimum, every `public` class and function) needs a proper docstring, with a brief description, as well as `@param` and `@return` tags.  See https://www.oracle.com/technetwork/articles/java/index-137868.html